### DISCUSSION TITLE
magick++

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#2204] [Help] [answered]
Hi,

Trying to read a gif animation using Magick++.

Magick::readImages(&imageList, file.c_str());

This indeed gives all the frames, only: first frame looks perfect, all further mis information. as if they're deltaframes that have not been overlayed to the first. how do I resolve that?‎

**Comment by snibgo** [comment, marked as answer]
Use coalesceImages(). See the sample program coalesceImages.cpp.

**Comment by ghost** [comment]
Oh indeed!

Thanks!
